### ISSUE TITLE
403 Forbidden at first login attempt - "The specified credentials are invalid"

### ISSUE THREAD

**Issue by wvmstr** [Microsoft/Bing-Maps-Fleet-Tracker#42]
"You can sign up for a new developer account at bingmapsportal.com" flashed briefly on the screen. But I cannot login,

I already have a developer account. 
Created a new key, tested the new key and it all seems to work with a simple test in the browser.

This is my second attempt and have now two fleet trackers in Azure, neither works.

Would it not be better if I was allowed to login and potentially make changes to the Bing maps key, rather than create a whole new application?

What other troubleshooting can I do?

**Comment by MustafaLeithy** [comment]
You can actually update any settings in your deployment, without creating a new one. See [Updating Deployments](https://github.com/Microsoft/Bing-Maps-Fleet-Tracker/blob/master/DEPLOYMENT.md#updating-an-existing-deployment).

It seems your issue is login related, and that is not tied to your Bing Maps Key. You will see the "You can sign up for a new developer account at bingmapsportal.com" message on the Bing Maps Control if you have not successfully logged in.

 Did you go to the [Microsoft Application Portal](https://apps.dev.microsoft.com/?deeplink=/appList) and create an application? Did you enter the application credentials into the portal?

**Comment by wvmstr** [comment]
Yes I did create an application, created a new secret word and entered both on the form.

I waited for the message "deployment successful" both on the deployment page and in Azure.

Did that twice now.

By not being able to update I meant that I cannot login (to the fleet tracker site) and make changes to my configuration. Of course that was when I thought that the error message displayed was relevant to why I cannot log in.

I will delete the first deployment.

When I started the deployment I entered the email address from my Azure account.  Is my password in the fleet tracker portal the same as my Azure account password, or is there a place in the App Service where I can find that password?

**Comment by MustafaLeithy** [comment]
> When I started the deployment I entered the email address from my Azure account.

This will be the only email that can log into the portal at first. Once you login with that email you can give access to other users. Make sure when you login into your portal, you are logging in with that email.

> Is my password in the fleet tracker portal the same as my Azure account password, or is there a place in the App Service where I can find that password?

The fleet tracker portal does not store your password. We use Azure Active Directory (AAD) as the login provider. Therefore your Azure login credentials are the same as the credentials you will use to sign in to your fleet tracker portal.

I am still not entirely sure what the issue with your login is. Are you being redirected to a login page? Does the login page successfully sign you in and redirect you back to the portal, or does it throw an error halfway through? Can you please attach a screenshot or describe what you see when you try and sign in to your fleet tracker portal?

**Comment by wvmstr** [comment]
[Image]




Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by wvmstr** [comment]
I am logged in to the Azure portal
in a new tab I browse to ltd4liberty.azurewebsites.net and hit enter
the loading icon is on the screen followed by a brief glimpse of the portal 
in the top right a red-ish box appears with "an error occurred while contacting the server"
then it asks me to log in (remember I am already logged in to the Azure portal)
click on the same email address I am already logged in with, enter password
the : " you don't have authorization to view the page" Error 403 Forbidden

I have a screen recording from the game bar that I can send you. Format is not accepted here on Github.

**Comment by wvmstr** [comment]
![image](https://user-images.githubusercontent.com/19780051/37875580-603913b6-300f-11e8-9fd3-b784767694c1.png)

**Comment by MustafaLeithy** [comment]
Do you receive a screen looking like this when you are redirected back to the portal?
![image](https://user-images.githubusercontent.com/25579739/37875809-631679cc-3044-11e8-9e6d-b3ce53fcef8e.png)

**Comment by wvmstr** [comment]
No. I get redirected to the 403 forbidden error page.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by wvmstr** [comment]
Here is another scenario:

log in to Azure Portal and I am looking at the dashboard;
from the same browser tab I navigate to the address of the fleet tracker
it says "loading"
then the window in the top right now says "not logged in"
it redirects me to "pick an account"

**Comment by MustafaLeithy** [comment]
The scenario you describe is working as intended. The only question is why does it not send you back to the portal after you have picked your account. Is the url of the 403 forbidden error page "api/users/accessdenied"?

**Comment by wvmstr** [comment]
Yes.


________________________________

**Comment by wvmstr** [comment]
It looks to me that the redirect back to the site is failing because it doesn't recognize I am logged in.

What say you Microsoft?

**Comment by MustafaLeithy** [comment]
What you are seeing means that AAD is refusing to grant your application a token. This probably means your Application Id or Secret were copied wrong, or that your application lacks the necessary permissions somehow.

Did you follow the README steps [for creating a Microsoft application](https://github.com/Microsoft/Bing-Maps-Fleet-Tracker/blob/master/DEPLOYMENT.md#creating-a-new-deployment)?

**Comment by wvmstr** [comment]
I had forgotten the redirect url on my last attempt.

Did another deployment with the redirect and added the app under converged applications and now I can log in.

Looking back I think I have to agree that the app under Converged Applications did the trick.

Might want to update the instructions so others don't make the same mistake.

**Comment by MustafaLeithy** [comment]
Great to hear. I will close the issue now, and update the documentation accordingly.

**Comment by larrywelch** [comment]
I'm having the exact same problem.  I attempted to use the Converged Applications, Azure AD App Registrations, and App Registration (Preview).  The Converged Applications indicates it won't be supported after May 2019 so I believe I need to use on of the Azure App Registration options.  In any case, I've gotten various results.  Using the Converged Application I was able to login but I was then told I didn't have permissions.  The Bing Map page showed but nothing else (no menu options on the left).  I then switched to the Azure App and in I then get 403 errors.


Any help will be greatly appreciated!
